pkg/dns/server/name_table.go

```diff
@@ -41,120 +41,122 @@ func BuildNameTable(cfg Config) *dnsProto.NameTable {
 	out := &dnsProto.NameTable{
 		Table: make(map[string]*dnsProto.NameTable_NameInfo),
 	}
-	for _, svc := range cfg.Node.SidecarScope.Services() {
-		var addressList []string
-		hostName := svc.Hostname
-		headless := false
-		for _, svcAddress := range svc.GetAllAddressesForProxy(cfg.Node) {
-			if svcAddress == constants.UnspecifiedIP {
-				headless = true
-				break
-			}
-			// Filter out things we cannot parse as IP. Generally this means CIDRs, as anything else
-			// should be caught in validation.
-			if !netutil.IsValidIPAddress(svcAddress) {
-				continue
+	for _, el := range cfg.Node.SidecarScope.EgressListeners {
```

Comment:
Is this the only change in this file?  What is the purpose of this change instead of accessing services directly from sidecar? The diff is difficult to parse

Reply:
yes it is. The .services is a single service per hostname already flattened. so 2 vips on a SE means you only end up with 1 breaking dual stack SE. 

The egress listener is not fully filtered like that - mirroring how LDS works